src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -166,37 +178,72 @@ export class EditorComponent implements OnInit {
       this.scaleFactor =  1;
     }
 
-    //  Canvas to draw mask, hidden.
+    // Canvas to draw mask, hidden.
     this.maskCanvas.nativeElement.width = imgWidth;
     this.maskCanvas.nativeElement.height = imgHeight;
     this.maskCtx = this.maskCanvas.nativeElement.getContext('2d');
     this.maskCtx.lineCap = this.maskCtx.lineJoin = 'round';
     this.maskCtx.strokeStyle = this.MAGENTA;
 
-    //  Canvas to show mask scaled.
+    // Canvas to show mask scaled.
     this.scaledCanvas.nativeElement.width = imgWidth * this.scaleFactor;
     this.scaledCanvas.nativeElement.height = imgHeight * this.scaleFactor;
     this.scaledCtx = this.scaledCanvas.nativeElement.getContext('2d');
 
-    //  Canvas to show Image (never changes unless user only wants to see Mask)
+    // Canvas to show Image (never changes unless user only wants to see Mask)
     this.imageCanvas.nativeElement.width = imgWidth * this.scaleFactor;
     this.imageCanvas.nativeElement.height = imgHeight * this.scaleFactor;
     this.imageCtx = this.imageCanvas.nativeElement.getContext('2d');
-    
+
     this.stageWidth = imgWidth * this.scaleFactor;
     this.stageHeight = imgHeight * this.scaleFactor;
 
-    //   Draws image non scaled on full canvas
+    // Canvas to paint cursor-overlay of brush size.
+    this.cursorCanvas.nativeElement.width = imgWidth * this.scaleFactor;
+    this.cursorCanvas.nativeElement.height = imgHeight * this.scaleFactor;
+
+    // Updates the drawn 'cursor' when the user's mouse moves.
+    this.maskTarget = document.querySelector('#mask-layer');
+    this.cursorTarget = document.querySelector('#cursor-layer');
+    this.cursorCtx = this.cursorTarget.getContext('2d');
+    this.maskTarget.addEventListener('mousemove', (e) => {
+      this.setCursorPosition(e);
+    }, false);
+    this.maskTarget.addEventListener('mouseout', () => {
+      this.cursorCtx.clearRect(0, 0, this.stageWidth, this.stageHeight);
+    }, false);
+
+    // Draws image non scaled on full canvas
     this.imageCtx.drawImage(this.image, 0, 0);
 
-    //  Only gets the image data from (0,0) to (width,height) of image.
+    // Only gets the image data from (0,0) to (width,height) of image.
     this.originalImageData = this.imageCtx.getImageData(0, 0, imgWidth, imgHeight);
 
     this.drawScaledImage();
     console.log('put imagedata');
   }
+
+
+  /* Handles cursor tracking and resizing. */
   
- /**
+  // Draws/Redraws 'cursor' at the current position of user's mouse.
+  setCursorPosition(e) {
+    // These are the coordinates used to paint.
+    this.cursorX = e.offsetX;
+    this.cursorY = e.offsetY;
+
+    this.cursorCtx.clearRect(0, 0, this.stageWidth, this.stageHeight);
+
+    this.cursorCtx.beginPath();
+    this.cursorCtx.arc(this.cursorX, this.cursorY,
```

Comment:
I think if you scale the ctx: 
`this.cursorCtx.scale(this.scaleFactor, this.scaleFactor);`
You can avoid having to set the radius to brushWidth * scaleFactor * .5. because it should automatically scale the radius for you and be the proper size. You may have to play around with this some, I'm not entirely sure.

Reply:
I think drawing a circle wasn't matching the brush stroke size even on un-scaled images. It was still about twice as big, which is why I needed to scale circle radius by .5